api/api.go

```diff
@@ -1,7 +1,12 @@
 package api
 
+//go:generate moq -out mockauth_test.go . AuthHandler
```

Comment:
Can I suggest generating the mock into a `mock` subdir? (eg. https://github.com/ONSdigital/dp-image-api/tree/develop/service/mock) I like the idea of generated sources being kept separate from authored sources.

Reply:
Moved all the mocks into their own sub directory, moved all interfaces from api.go and handlers.go to interfaces.go. Changed the _test packages to api_test to avoid the circular imports. (looks a lot better structured like this)